valhalla/tyr/route_summary_cache.h

```diff
@@ -0,0 +1,96 @@
+#ifndef VALHALLA_TYR_ROUTE_SUMMARY_CACHE_H_
+#define VALHALLA_TYR_ROUTE_SUMMARY_CACHE_H_
+
+#include "proto/directions.pb.h"
+#include "proto/options.pb.h"
+#include "proto/trip.pb.h"
+#include "proto_conversions.h"
+
+namespace valhalla {
+namespace tyr {
+
+struct NamedSegment {
```

Comment:
here I'd be careful. when we have headers outside of the `valhalla` directory then that usually means they're considered private and only used for the compilation but not publicly installed/distributed.

Reply:
Keeping on header, but as private member of route_summary_cache class

Reply:
hm, can you elaborate? I still see that this is in `./valhalla/tyr` directory, so it's still public as it'll be installed. what am I missing?